src/Agent.Worker/ExecutionContext.cs

```diff
@@ -244,23 +244,30 @@ public void InitializeJob(JobRequestMessage message)
 
             // Initialize the environment.
             Endpoints = message.Environment.Endpoints;
-            Variables = new Variables(HostContext, message.Environment.Variables);
+            List<string> warnings;
+            Variables = new Variables(HostContext, message.Environment.Variables, out warnings);
 
             // Initialize the job timeline record.
             // the job timeline record is at order 1.
             InitializeTimelineRecord(message.Timeline.Id, message.JobId, null, ExecutionContextType.Job, message.JobName, 1);
-        }        
+
+            // Log any warnings.
+            foreach (string warning in (warnings ?? new List<string>()))
```

Comment:
I try to avoid additional scopes where unnecessary. I feel it stands in the way of readability. I don't know why the built-in IEnumerable extension methods don't include a `CoalesceWithEmpty()` or `OrEmpty()` method as a succinct way to avoid null checks before iterating.